pkg/operators/klusterlet/controllers/ssarcontroller/klusterlet_ssar_controller_test.go

```diff
@@ -0,0 +1,331 @@
+				testinghelper.NamedCondition(hubConfigSecretDegraded, "HubKubeConfigError", metav1.ConditionTrue),
+			},
+		},
+		{
+			name: "Unauthorized",
+			object: []runtime.Object{
+				newSecretWithKubeConfig(helpers.BootstrapHubKubeConfig, "test", newKubeConfig(apiServerHost)),
+				newSecretWithKubeConfig(helpers.HubKubeConfig, "test", newKubeConfig(apiServerHost)),
+			},
+			allowToOperateManagedClusters:      false,
+			allowToOperateManagedClusterStatus: false,
+			allowToOperateManifestWorks:        false,
+			klusterlet:                         newKlusterlet("testklusterlet", "test", "cluster1"),
+			expectedConditions: []metav1.Condition{
+				testinghelper.NamedCondition(bootstrapSecretDegraded, "BootstrapSecretUnauthorized", metav1.ConditionTrue),
+				testinghelper.NamedCondition(hubConfigSecretDegraded, "HubKubeConfigUnauthorized", metav1.ConditionTrue),
+			},
+		},
+		{
+			name: "Operator functional",
+			object: []runtime.Object{
+				newSecretWithKubeConfig(helpers.BootstrapHubKubeConfig, "test", newKubeConfig(apiServerHost)),
+				newSecretWithKubeConfig(helpers.HubKubeConfig, "test", newKubeConfig(apiServerHost)),
+			},
+			allowToOperateManagedClusters:      true,
+			allowToOperateManagedClusterStatus: true,
+			allowToOperateManifestWorks:        true,
+			klusterlet:                         newKlusterlet("testklusterlet", "test", "cluster1"),
+			expectedConditions: []metav1.Condition{
+				testinghelper.NamedCondition(bootstrapSecretDegraded, "BootstrapSecretFunctional", metav1.ConditionFalse),
+				testinghelper.NamedCondition(hubConfigSecretDegraded, "HubConfigSecretFunctional", metav1.ConditionFalse),
+			},
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			controller := newTestController(c.klusterlet, c.object...)
+			syncContext := testinghelper.NewFakeSyncContext(t, c.klusterlet.Name)
+
+			response.allowToOperateManagedClusters = c.allowToOperateManagedClusters
+			response.allowToOperateManagedClusterStatus = c.allowToOperateManagedClusterStatus
+			response.allowToOperateManifestWorks = c.allowToOperateManifestWorks
+
+			err := controller.controller.sync(context.TODO(), syncContext)
+			if err != nil {
+				t.Errorf("Expected no error when update status: %v", err)
+			}
+
+			// wait util goroutine is done
+			for controller.controller.inSSARChecking(c.klusterlet.Name) {
+				time.Sleep(time.Second * 1)
+			}
+
+			operatorActions := controller.operatorClient.Actions()
+
+			testinghelper.AssertEqualNumber(t, len(operatorActions), 2)
+			testinghelper.AssertGet(t, operatorActions[0], "operator.open-cluster-management.io", "v1", "klusterlets")
+			testinghelper.AssertAction(t, operatorActions[1], "update")
+			testinghelper.AssertOnlyConditions(t, operatorActions[1].(clienttesting.UpdateActionImpl).Object, c.expectedConditions...)
+		})
+	}
+}
+
+func TestLocker(t *testing.T) {
```

Comment:
@qiujian16  Plz has a view. This would depict klusterlets would not affect each other.